sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -4129,6 +4129,37 @@ class Dataset[T] private[sql](
     new DataFrameWriterV2[T](table, this)
   }
 
+  /**
+   * Create a [[DataFrameWriterV2]] for mergeInto action.
+   *
+   * Scala Examples:
+   * {{{
+   *   spark.table("source")
+   *     .mergeInto("target")
+   *     .on($"source.id" === $"target.id")
+   *     .whenMatched($"salary" === 100)
+   *     .delete()
+   *     .whenNotMatched()
+   *     .insertAll()
+   *     .whenNotMatchedBySource($"salary" === 100)
```

Comment:
Is `whenNotMatchedBySource` simply `whenNotMatched` plus a source condition? If so, why it isn't `whenNotMatched($"salary" === 100)` like `whenMatched` with condition?